storey/steps/groupbykey.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional, Union
+
+from storey.dtypes import Event
+from storey.flow import _Batching
+
+
+class GroupByKey(_Batching):
```

Comment:
In my opinion it's better to just extend this functionality to `Batch`, rather than a new Step. also it's not really `GroupByKey` more of `BatchByKey`

Reply:
Then maybe `Batch` should change to Group with an optional key, maybe it should follow a more standard name for this type of operation. I've never called a batch function, but i did call GroupBy.

What do you think?

Reply:
GroupBy is sort of a finite/reduce operation it's grouping the entire dataset, while Batch/BatchByKey just batches a bunch of events based on some logic(max events/timeout) and emits them forward possible by key.